src/components/App.js

```diff
@@ -194,17 +194,20 @@ export default class App extends React.Component {
 
   pause() {
     // completely stop the audio element
-    if (this.state.playing) {
-      this._player.src = '';
+    if (!this.state.playing) return Promise.resolve();
+
+    return new Promise(resolve => {
       this._player.pause();
       this._player.load();
 
       this.setState({
         playing: false,
         pausing: false
+      }, () => {
+        SUB.stop();
+        resolve();
       });
-      SUB.stop();
-    }
+    })
```

Comment:
I made this return a promise in order to resolve the "'await' has no effect on the type of this expression" warning in this line:

https://github.com/freeCodeCamp/coderadio-client/blob/4bfaa7ba8cc63d28b2aab18a5d39d014bfb733ab/src/components/App.js#L147